### ISSUE TITLE
[Bug] Links in index are getting broken during second gzip-enabled generation

### ISSUE THREAD

**Issue by kiler129** [prestaconcept/PrestaSitemapBundle#197]
While generating sitemap section-by-section I stepped into rather interesting bug. 

## Problem
Let's say you have two sections: `static` and `dynamic`. You want to generate both separately so you run:

```
bin/console presta:sitemaps:dump --base-url 'https://www.example.com/sitemap/' --gzip --section static
bin/console presta:sitemaps:dump --base-url 'https://www.example.com/sitemap/' --gzip --section dynamic
```

After the first command run you will get:
 - `sitemap.xml`
 - `sitemap.static.xml.gz`

The `sitemap.xml` will contain the following contents:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<sitemapindex xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/siteindex.xsd"
              xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
    <sitemap>
        <loc>https://www.example.com/sitemap/sitemap.static.xml.gz</loc>
        <lastmod>2019-04-29T09:55:50-05:00</lastmod>
    </sitemap>
</sitemapindex>
```

**The problem arises after the second command is run.** While the folder correctly contains the following files:
 - `sitemap.xml`
 - `sitemap.static.xml.gz`
 - `sitemap.dynamic.xml.gz`

the index itself is broken:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<sitemapindex xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/siteindex.xsd"
              xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
    <sitemap>
        <loc>https://www.example.com/sitemap/sitemap.dynamic.xml.gz</loc>
        <lastmod>2019-04-29T09:57:20-05:00</lastmod>
    </sitemap>              
    <sitemap>
        <loc>https://www.example.com/sitemap/sitemap.static.xml</loc>
        <lastmod>2019-04-29T09:55:50-05:00</lastmod>
    </sitemap>
</sitemapindex>
```

## Reason
It seems like the dumper "forgets" that the sitemaps read from previously created index are gzip-ed:
https://github.com/prestaconcept/PrestaSitemapBundle/blob/5c9f41e6fee40477f256250a1f1859b2208ff4b7/Service/Dumper.php#L170-L174

## Solution
The workaround as of now is to never run the command with `--section` - when run without it (i.e. generating all sections at once) it will respect the `--gzip` flag and actually re-create the url-set. While it's not a solution it at least allows for generation of proper index :)

Before I offer a PR I'm trying to understand one crucial design decision here: why links are re-generated at all? For me it will be logical to simply copy these links?

The simple fix without any BC break will be to just pass the file extension (so that `sitemapFilePrefix` is still respected etc) to the `\Presta\SitemapBundle\Service\Dumper::newUrlset`.

I can offer a PR if you like the solution.

**Comment by kiler129** [comment]
@yann-eugone This bug is really kicking our asses :D I will be happy to fix it if you're fine with the suggested solution.

**Comment by yann-eugone** [comment]
Released in [`v2.1.0`](https://github.com/prestaconcept/PrestaSitemapBundle/releases/tag/v2.1.0)
